### PR TITLE
Add pre_save recalculation

### PR DESCRIPTION
**Pull request by Jokeswar**
#92 `total_score` is now read-only. On save from dajngo admin  `total_score` is recalculated. It takes into account changes from `score`, `penalty`, `review_score`. If the `review_message` is changed `review_score` will not be updated(i.e. will not be recalculated based on `review_message`), and there will be no impact on `total_score`.

### CODE REVIEW

**Comment by Jokeswar** [comment]
Let's say a TA gives a review with the normal review button in `submission/<pk>` :
```
+1.0: great
```
This [part](https://github.com/vmck/acs-interface/blob/master/interface/views.py#L110-L121) here gets the `review_score` as `1`. Let's say that the `total_score` is 101 (100 from tests, 1 from `review_score`). Now if we change the review message to:
```
+2.0: good
```
in django admin and hit save, `total_score` will still be 101 (100 from tests, 1 from `review_score`). What I want to say is that `review_message` is not reprocessed because [this](https://github.com/vmck/acs-interface/blob/master/interface/models.py#L117-L126) only takes the `review_score` without looking at `review_message`. @mgax

**Comment by mgax** [comment]
> What I want to say is that `review_message` is not reprocessed because [this](https://github.com/vmck/acs-interface/blob/master/interface/models.py#L117-L126) only takes the `review_score` without looking at `review_message`

Then why don't we move [this](https://github.com/vmck/acs-interface/blob/master/interface/views.py#L110-L121) to [here](https://github.com/vmck/acs-interface/blob/master/interface/models.py#L117-L126)?

**Comment by Jokeswar** [comment]
I'll do that, and I believe we should make `review_score` read-only too.

**Comment by mgax** [comment]
> I believe we should make `review_score` read-only too.

I agree.

File: interface/views.py [hunk 1]
```diff
@@ -109,18 +109,8 @@ def homepage(request):
...
-        re.MULTILINE,
-    )
-    log.debug('Marks found: ' + str(marks))
-
-    review_score = sum([decimal.Decimal(mark) for mark in marks])
-
-    submission.review_score = review_score
-    submission.total_score = submission.calculate_total_score()
     submission.review_message = request.POST['review-code']
+    submission.total_score = submission.calculate_total_score()
```
Comment (mgax) [inline interface/views.py]: Since `calculate_total_score` is called in the pre-save hook, this is a bit redundant, no?

**Pull request merged**